OpenRA.Mods.RA2/Traits/AttackInfect.cs

```diff
@@ -0,0 +1,105 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.RA2.Activities;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("Move onto the target then execute the attack.")]
+	public class AttackInfectInfo : AttackFrontalInfo, Requires<MobileInfo>
+	{
+		[Desc("Range of the final joust of the infector.")]
+		public readonly WDist JoustRange = WDist.Zero;
+
+		[Desc("Conditions that last from start of the joust until the attack.")]
+		[GrantedConditionReference]
+		public readonly string JoustCondition = "jousting";
+
+		[FieldLoader.Require]
+		[Desc("How much damage to deal.")]
+		public readonly int Damage = 0;
+
+		[FieldLoader.Require]
+		[Desc("How often to deal the damage.")]
+		public readonly int DamageInterval = 0;
+
+		[Desc("Damage types for the infection damage.")]
+		public readonly BitSet<DamageType> DamageTypes = default(BitSet<DamageType>);
+
+		[Desc("If an external actor delivers more damage than this value, the infector is killed immediately.",
+			"Use -1 to never kill the infector.")]
+		public readonly int SuppressionDamageThreshold = -1;
```

Comment:
What is the usecase for all the `supression` properties here, I don't remember any of that in RA2?